src/Modal/Modal.tsx

```diff
@@ -45,13 +47,29 @@ export const Modal: FC<ModalProps> = ({
   portalContainer = document.body,
 }) => {
   const modalRef = useRef<HTMLDivElement>(null)
+  const hasOpened = useRef(false)
+  const [opened, setOpened] = useState(false)
 
   useBodyScrollLock({ node: modalRef.current, showModal })
 
+  useEffect(() => {
+    if (modalRef.current === null) return
+
+    if (!hasOpened.current && showModal) {
```

Comment:
`showModal` in the end is needed because we want to trigger it only on the first click, without it it will append it at first render

Reply:
😢